README.md

```diff
@@ -1,105 +1,152 @@
+  - `INSERT` and `UPDATE` only work for singleton key `ZSET` tables if they have the
+  priority column
+  - non-singleton non-scalar tables must have an array type for the second column
 
-    PATH=/usr/local/pgsql91/bin/:$PATH make USE_PGXS=1
-    sudo PATH=/usr/local/pgsql91/bin/:$PATH make USE_PGXS=1 install
+### Pushdowning
 
-(assuming you have PostgreSQL 9.1 in /usr/local/pgsql91).
+Not supported, there is no common calculations in Redis.
 
-Make necessary changes for 9.2 and later.
+### Notes about features
 
-You will need to have the right branch checked out to match the PostgreSQL
-release you are building against, as the FDW API has changed from release
-to release.
+Also see [Limitations](#limitations)
+
+Supported platforms
+-------------------
 
-Dave has tested the original on Mac OS X 10.6 only, and Andrew on Fedora and
+`redis_fdw` was developed on Linux and Mac OS X and should run on any
+reasonably POSIX-compliant system.
+[Dave](#license-and-authors) has tested the original on Mac OS X 10.6 only, and [Andrew](#license-and-authors) on Fedora and
 Suse. Other *nix's should also work.
 Neither of us have tested on Windows, but the code should be good on MinGW.
 
-Limitations
------------
+Installation
+------------
 
-- There's no such thing as a cursor in Redis in the SQL sense,
-  nor MVCC, which leaves us
-  with no way to atomically query the database for the available keys
-  and then fetch each value. So, we get a list of keys to begin with,
-  and then fetch whatever records still exist as we build the tuples.
+No deb or rpm packages are avalillable.
 
-- We can only push down a single qual to Redis, which must use the
-  TEXTEQ operator, and must be on the 'key' column.
+### Source installation
 
-- There is no support for non-scalar datatypes in Redis
-  such as lists, for PostgreSQL 9.1. There is such support for later releases.
+#### Prerequisites:
+- A Redis database accesable from PostgreSQL server.
+- Local Redis *only* if you need `redis_fdw` testing.
+- [Hiredis C interface](https://github.com/redis/hiredis) installed
+on your system. You can checkout the `hiredis` from github or it might be available in [rpm or deb packages for your OS](https://pkgs.org/search/?q=hiredis).
+- PostgreSQL development package. For Debian or Ubuntu: `apt-get install postgresql-server-dev-XX`, where `XX` matches your postgres version, i.e. `apt-get install postgresql-server-dev-15`
 
-- Redis has acquired cursors as of Release 2.8. This is used in all the
-  mainline branches from REL9_2_STABLE on, for operations which would otherwise
-  either scan the entire Redis database in a single sweep, or scan a single,
-  possible large, keyset in a single sweep. Redis Releases prior to 2.8 are
-  maintained on the REL9_x_STABLE_pre2.8 branches.
+#### Build and install on OS
 
-- Redis cursors have some significant limitations. The Redis docs say:
+Add a directory of `pg_config` to PATH and build and install `regis_fdw`.
```

Comment:
The phrasing here needs to be reworked as it only just makes sense as written, and may be difficult for some to understand. Also, redid_fdw is misspelt.

Reply:
Changed in https://github.com/pg-redis-fdw/redis_fdw/pull/37/commits/01bce9dfdce3ebcfe0dc6dc5e5ad05abede8315a, please review the new variant.